### ISSUE TITLE
Render text into textures

### ISSUE THREAD

**Issue by TheRawMeatball** [bevyengine/bevy#1159]
Currently, text is rendered using the sprite sheet pipeline. This has various issues:

- Breaks code encapsulation: They are similar, but linking them makes it harder to develop features for text rendering and vice versa.
- Is really slow: (see #1122)
- Blocks #1158 

I propose text rendering is done using a specialized pipeline, rendering the text into a texture that can stay on the GPU improving performance, and enabling text to use the same render pipeline the rest of the UI systems use.

As an alternative, optimization could be done elsewhere, and spritesheets and text could stay linked, but neither would allow the approach in #1158 (or a possible reimplementation).

**Comment by PabloMansanet** [comment]
I'm very interested in this and I've done some work on text as part of #1122, so I'm happy to give this a shot, or to help out if you've already started work on it. I'd appreciate some mentoring around the rendering side, which I'm less familiar with.

**Comment by TheRawMeatball** [comment]
I tried and failed to implement this myself. I'm not very familiar with the rendering system either, but as far as I know, there isn't a way to render to a texture yet. I would love to be proven wrong on this feature being blocked on another, but I think the best course of action would be to first implement a generic way to render stuff into textures, which could probably use its own issue.

**Comment by PabloMansanet** [comment]
Right, I see. Going to give that a shot. If nothing else, it will be a great excuse to learn about the rendering pipeline :)

**Comment by tigregalis** [comment]
For rendering to a texture, consider:

- https://github.com/dmitriy-shmilo/sio2/blob/master/src/render.rs#L45-L64
- https://github.com/bevyengine/bevy/blob/master/crates/bevy_sprite/src/texture_atlas_builder.rs#L83-L84
